### ISSUE TITLE
Training using multiple GPUs affects total loss calculation (regularisation loss part increases)

### ISSUE THREAD

**Issue by loztop** [fchollet/keras#7178]
Hi,

Thanks for the great Keras project. I just wanted to report a small problem I have come across.

When training using multiple gpus, setup using https://github.com/kuza55/keras-extras/blob/master/utils/multi_gpu.py , the weight decay part (for example l2 weight decay) of the total loss calculation is added multiple times for each gpu batch split.

I.e. The more gpus used to split the batch -> the higher the loss ! I'm guessing somewhere each gpu is separately adding the same regularization loss to the model's total loss.

The summing of all extra (e.g. regularization) losses happens in https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L846-L847

For now my fix is to just divide the weight_decay by the number of gpus used (e.g. 0.0001/16), to ensure that the loss calculation does not depend on the number of gpus.

Is there a better/cleaner solution ?

Thanks

**Comment by ajeetksingh** [comment]
@loztop did it help, dividing the weight_decay by the number of gpus used? i think i am also stuck at this issue. Thanks.

**Comment by look4pritam** [comment]
Is it solved? I am also facing the same problem. Dividing weight decay by number of GPUs solves problem little bit but not completely.
